src/main/java/com/google/cloud/tools/app/Action.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.app;
+
+import com.google.cloud.tools.app.Option.Type;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+/**
+ * Contains common members and methods to all {@link Action} implementations.
+ */
+// TODO(joaomartins): Come up with a better name for this. Action too overloaded.
```

Comment:
I was thinking more like Operation? I would think it's OK to have generic names, they would be scoped to our library no?

AppOperation is another possibility (taking on your CloudAction). I'd live to avoid more "GCloud" branding.

@elharo what do you think?